otx/cli/tools/build.py

```diff
@@ -49,29 +61,85 @@ def main():
     """Main function for model or backbone or task building."""
 
     args = parse_args()
+    args.task = args.task.upper() if args.task is not None else args.task
 
     builder = Builder()
 
-    # Build with task_type -> Create User workspace
     otx_root = get_otx_root_path()
 
-    if args.task and args.task.upper() in SUPPORTED_TASKS:
+    # Auto configuration & Datumaro Helper
+    # First, find the format of train dataset and import it by using DatasetManager
+    # Second, find the task type
+    # Third, check the whether val_data_roots is None or not None
+    # If val_data_roots is None, auto_split() is trigerred.
+    # Forth, save the dataset with it's configuraiton(.yaml) file
+    is_autoconfig_enabled = args.train_data_roots
+    if is_autoconfig_enabled:
+        print("[*] Auto-configuration is enabled !!")
+        train_data_format = DatasetManager.get_data_format(args.train_data_roots)
+        print(f"[*] Train data format: {train_data_format}")
+
+        train_datumaro_dataset = DatasetManager.import_dataset(
+            data_root=args.train_data_roots, data_format=train_data_format
+        )
+
+        splitted_dataset = {}
+        splitted_dataset["train"] = DatasetManager.get_train_dataset(train_datumaro_dataset)
+
+        config_manager = ConfigManager()
+        train_task_type = config_manager.get_task_type(train_data_format)
+        print(f"[*] Train task type: {train_task_type}")
+        if args.task:
+            assert (
+                args.task == train_task_type
+            ), f"Dataset format({train_data_format}) can't be used for {args.task} task."
+
+        # Overwrite the args.task to train_task_type to select default template
+        args.task = train_task_type if args.task is None else args.task
+
+        # Create workspace
```

Comment:
In fact, from here, it doesn't have to belong to the if autoconfig statement, right?
I think lines 100-135 have little to do with auto-config.
What do you think?